admin-dev/themes/new-theme/js/pages/order/view/order-view-page.js

```diff
@@ -337,4 +285,73 @@ export default class OrderViewPage {
   getActivePage() {
     return $(OrderViewPageMap.productsTablePagination).find('.active span').get(0);
   }
+
+  refreshProductsList(orderId) {
+    $(OrderViewPageMap.refreshProductsListLoadingSpinner).show();
+
+    const $tablePagination = $(OrderViewPageMap.productsTablePagination);
+    const numRowsPerPage = $tablePagination.data('numPerPage');
+    const initialNumProducts = $(OrderViewPageMap.productsTableRows).length;
+    let currentPage = parseInt($(OrderViewPageMap.productsTablePaginationActive).html(), 10);
+
+    $.ajax(this.router.generate('admin_orders_get_products', {orderId}))
+        .done((response) => {
+          // Delete previous product lines
+          $(OrderViewPageMap.productsTable).find(OrderViewPageMap.productsTableRows).remove();
+
+          $(OrderViewPageMap.productsTable + ' tbody').prepend(response);
+
+          $(OrderViewPageMap.refreshProductsListLoadingSpinner).hide();
+
+          const newNumProducts = $(OrderViewPageMap.productsTableRows).length;
+          const newPagesNum = Math.ceil(newNumProducts / numRowsPerPage);
+
+          this.orderProductRenderer.updateNumProducts(newNumProducts);
+          this.orderProductRenderer.updatePaginationControls();
+
+          let numPage = 1;
+          let message = '';
+          // Display alert
+          if (initialNumProducts > newNumProducts) { // product deleted
+            message = (initialNumProducts-newNumProducts === 1) ?
+                window.translate_javascripts['The product was successfully removed.'] :
+                window.translate_javascripts['[1] products were successfully removed.']
+                    .replace('[1]', (initialNumProducts-newNumProducts))
+            ;
+
+            // Set target page to the page of the deleted item
+            numPage = (newPagesNum === 1) ? 1 : currentPage;
+          }
+          else if (initialNumProducts < newNumProducts) { // product added
+            message = (newNumProducts - initialNumProducts === 1) ?
+                window.translate_javascripts['The product was successfully added.'] :
+                window.translate_javascripts['[1] products were successfully added.']
+                    .replace('[1]', (newNumProducts-initialNumProducts))
+            ;
+
+            // Move to first page to see the added product
+            numPage = 1;
+          }
+
+          $.growl.notice({
+            title: '',
+            message: message,
```

Comment:
if message is empty, what is the behavior ?

Reply:
an empty growl is displayed. I updated the code to test if message is empty